openwisp_users/admin.py

```diff
@@ -185,6 +185,15 @@ class Meta(BaseUserCreationForm.Meta):
     class Media:
         js = ('admin/js/jquery.init.js', 'openwisp-users/js/addform.js')
 
+    def clean(self):
```

Comment:
I don't remember having asked that we do this in the admin form, why do this in the admin when we can do it in the main model? We always implement these kind of operations in the models, because this validation logic can be shared across different views (eg: admin, API views, custom views).

https://github.com/openwisp/openwisp-users/blob/master/openwisp_users/base/models.py#L143-L147

Reply:
I thought of implementing it "closer to the user". But you are right on this one, I will take care of this from next time. Will make the change soon.